### ISSUE TITLE
Dataset splitting for Python data reader

### ISSUE THREAD

**Issue by timmoon10** [LLNL/lbann#1125]
In most data readers, the `validation_percent` field can be set to split off a dataset into a training set and validation set. However, it doesn't do anything with the Python data reader.

A quick fix shouldn't be hard. However, there is a larger question of how to generalize this so we aren't rigidly forced into the train/validation/test split (for example, if you need multiple distinct validation sets for LTFB, or if you don't need a training set for unit testing).
